models/migrations/v101.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"time"
+
+	"github.com/go-xorm/xorm"
+)
+
+func changeSomeColumnsLengthOfExternalLoginUser(x *xorm.Engine) error {
+	type ExternalLoginUser struct {
+		ExternalID        string                 `xorm:"pk NOT NULL"`
+		UserID            int64                  `xorm:"INDEX NOT NULL"`
+		LoginSourceID     int64                  `xorm:"pk NOT NULL"`
+		RawData           map[string]interface{} `xorm:"TEXT JSON"`
+		Provider          string                 `xorm:"index VARCHAR(25)"`
+		Email             string
+		Name              string
+		FirstName         string
+		LastName          string
+		NickName          string
+		Description       string
+		AvatarURL         string
+		Location          string
+		AccessToken       string `xorm:"TEXT"`
```

Comment:
@lunny I've seen that in some cases all the columns are specified, and in others only the changes. In some of them, the changes + the ID column. What should be done here? Is it irrelevant?

Reply:
@guillep2k Since `Sync2` will never drop columns, both OK. But I think only `ID` and changed columns here will be better.

Reply:
So should I keep just `ExternalID`, `LoginSourceID` and three changed columns?

Reply:
You need to keep here only changed columns

Reply:
@lafriks how about those two IDs?

Reply:
If they are not changed than no need for them here imho

Reply:
so I don't need these ID columns even if they are primary key? I can fix that

Reply:
For this I don't think you need id